### ISSUE TITLE
Wait.until! always return true

### ISSUE THREAD

**Issue by rsporny** [MYOB-Technology/wait_until#1]
Hi, until! method always return true, look:
`def until!(args={}, &block)
  until_true!(args) do
    block.call
    true
  end
end`

**Comment by dueckes** [comment]
@rsporny The intention of `until!` is to wait until the provided block evaluates without error.

Consider this contrived example:
```ruby
Wait.until! { raise "Not finished" if rand(10) < 5  }
```

Are you suggesting you'd like `until!` to return the result of the `block` call?

**Comment by rsporny** [comment]
Yes, I thought that `until!` is just an alias for `until_true!`.
However, if your intention was slightly different it's totally ok.

**Comment by phuongnd08** [comment]
This is very odd. Wait.until! should be equivalent to Wait.until_true!
If you are talking about Wait.until finished without error, you should make it differnt: Wait.until_no_error {}
I would say most developers will be deceived by this behavior.

**Comment by phuongnd08** [comment]
@dueckes I have sent this PR: https://github.com/MYOB-Technology/wait_until/pull/3
I think this would stop deceiving devs with minimal side effect. People who have accustomed to until! can just stay on the old version or switch to new method until_no_errors!. I think it's better to not introduce Wait.until! again as it will beak either sides expectation.
